### PR TITLE
Add support for Baremetal provider

### LINKED ISSUE #6758: Add support for Baremetal provider
As a KKP I want to create a bare-metal cluster from the dashboard a

some notes regarding the Tinkerbell: 
* A Machine Deployment object will represent one and only bare-metal node
* A Machine Deployment has a reference to the bare-metal node (hardware object) - Specify the name and namespace of the hardware object

### PR DESCRIPTION
**Pull request by Waseem826**
**What this PR does / why we need it**:
This PR adds support for baremetal provider.

### Create Cluster Wizard

**Provider Step:**

screenshot-localhost_8000-2024 08 20-19_23_30

**Settings Step:**

screenshot-localhost_8000-2024 08 20-19_24_00

**Node Settings Step:**

screenshot-localhost_8000-2024 08 20-19_25_30

### Preset
**Create Preset Dialog:**

screenshot-localhost_8000-2024 08 20-19_26_06

**Which issue(s) this PR fixes**:

Fixes #6758 

**What type of PR is this?**

/kind feature
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add support for Baremetal provider.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: modules/web/src/app/cluster/details/cluster/node-list/template.html [hunk 1]
```diff
@@ -424,6 +424,21 @@
...
+            <ng-container *ngIf="element.spec.cloud.baremetal?.tinkerbell">
+              <km-property *ngIf="element.spec.cloud.baremetal.tinkerbell.hardwareRef?.Name">
+                <div key>Hardware Name</div>
+                <div value>{{ element.spec.cloud.baremetal.tinkerbell.hardwareRef?.Name }}</div>
+              </km-property>
+              <km-property *ngIf="element.spec.cloud.baremetal.tinkerbell.hardwareRef?.Namespace">
+                <div key>Hardware Namespace</div>
+                <div value>{{ element.spec.cloud.baremetal.tinkerbell.hardwareRef?.Namespace }}</div>
+              </km-property>
+              <km-property>
```
Comment (ahmadhamzh) [inline modules/web/src/app/cluster/details/cluster/node-list/template.html:437]: do we need a `ngIf` directive also here ?

File: modules/web/src/app/node-data/basic/provider/baremetal/component.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+  static Ready(os?: OperatingSystem) {
+    return `${this.getOSMessage(os)} System Image`;
+  }
+
+  static Empty(os?: OperatingSystem) {
+    return `No ${this.getOSMessage(os)} System Images Available`;
+  }
+
+  private static getOSMessage(os?: OperatingSystem): string {
```
Comment (ahmadhamzh) [inline modules/web/src/app/node-data/basic/provider/baremetal/component.ts:58]:
```suggestion
  private static _getOSMessage(os?: OperatingSystem): string {
```

File: modules/web/src/app/cluster/details/cluster/node-list/template.html (hunk 1 above)
Comment (Waseem826) [reply modules/web/src/app/cluster/details/cluster/node-list/template.html:437]: No, OS Image URL is a mandatory field so value will be there.

**Review by ahmedwaleedmalik — APPROVED** [review]
/hold
I haven't reviewed the code, will leave that up to @ahmadhamzh. From the screenshots, the UI looks good to me 👍

File: modules/web/src/app/node-data/basic/provider/baremetal/component.ts (hunk 2 above)
Comment (Waseem826) [reply modules/web/src/app/node-data/basic/provider/baremetal/component.ts]: Fixed.

**Pull request merged**